src/databases/Curve3D/avtCurve3DFileFormat.C

```diff
@@ -0,0 +1,1122 @@
+//  Creation:   October 28, 2005 
+//
+// ****************************************************************************
+
+double 
+avtCurve3DFileFormat::GetTime(void)
+{
+    return curveTime;
+}
+
+
+// ****************************************************************************
+//  Method: avtCurve3DFileFormat::GetCycle
+//
+//  Purpose: Return the cycle associated with this curve file
+//
+//  Programmer: Kathleen Bonnell 
+//  Creation:   October 28, 2005 
+//
+// ****************************************************************************
+
+int 
+avtCurve3DFileFormat::GetCycle(void)
+{
+    return curveCycle;
+}
+
+// ****************************************************************************
+//  Method: avtCurve3DFileFormat::GetAuxiliaryData
+//
+//  Purpose: Manage materials
+//
+//  Programmer: Olivier Cessenat
+//  Creation:   October 5, 2023
+//
+// ****************************************************************************
+
+void *
+avtCurve3DFileFormat::GetAuxiliaryData(const char *var, const char *type, void *,DestructorFunction &df)
+{
+    void *retval = 0;
+#ifdef MDSERVER
+    return retval;
+#endif
+#ifdef PARALLEL
+    if (PAR_Rank() != 0)
+        return retval ;
+#endif
+    if(strcmp(type, AUXILIARY_DATA_MATERIAL) == 0)
+    {
+        if (strcmp(var, "segments") == 0)
+        {
+            avtMaterial *mat = NULL;
+            int ndims = 1 ;
+            int dims[3];
+            int nelements = 0 ;
+            size_t i,j,k ;
+            size_t nmats = curves.size() ;
+            // Gets the number of nodes and elements :
+            for (i=0 ; i<nmats; i++)
+            {
+                vtkUnstructuredGrid *rg = curves[i] ;
+                vtkPoints *lpts = rg->GetPoints() ;
+                int npl = lpts->GetNumberOfPoints();
+                nelements += (npl-1) ;
+            }
+            dims[0] = nelements ; 
+            int *matnos = new int[nmats];
+            for (i=0; i<nmats; i++)
+                matnos[i] = i+1;
+
+            char **names = new char *[nmats];
+            for (i=0; i<nmats; i++)
+            {
+                names[i] = new char[strlen(curveNames[i].c_str())+1+4];
+                sprintf(names[i], "%s", ("mat_" + curveNames[i]).c_str());
+            }
+
+            int *matlist = new int[dims[0]];
+            j = -1;
+            for (i=0 ; i<nmats; i++)
+            {
+                vtkUnstructuredGrid *rg = curves[i] ;
+                vtkPoints *lpts = rg->GetPoints() ;
+                int npl = lpts->GetNumberOfPoints();
+                for (k=1; k< npl; k++)
+                    matlist[j+k] = 1+i ;
+                j += (npl - 1) ;
+            }
+#if INTERACTIVEPLOT
+            fprintf(stdout,"nmats=%ld\n",nmats);
+            for (i=0 ; i<nmats; i++)
+                fprintf(stdout,"\ti=%ld, no=%d, name=%s\n", i, matnos[i], names[i]);
+#endif
+            
+            mat = new avtMaterial(nmats,matnos,names,ndims,dims,0,matlist,
```

Comment:
I am not sure what you are using materials for in this case? Are you using it to provide a way to break a single curve object into different pieces?

Reply:
Exactly. This way a segmented curve is easy to visualize.